### ISSUE TITLE
Support named arguments when instantiating records via injector

### ISSUE THREAD

**Issue by GuySartorelli** [silverstripe/silverstripe-framework#11048]
Currently passing in named arguments to be used in use of injector (i.e. `MyClass::create('normal-arg', namedArg: 'value');`) will result in silently passing the named argument to the wrong parameter. This is done intentionally because we use the splat operator to unpack arguments to be passed to constructors, and that operator throws an error for any string keys - so we have to pass through only the values (see [`InjectionCreator::create()`](https://github.com/silverstripe/silverstripe-framework/blob/a1eee2a9ef78d04fa6558a89780b1ba0d72c5844/src/Core/Injector/InjectionCreator.php#L32-L35)).

We could change this to support named arguments in injection like so:

```diff
-// Ensure there are no string keys as they cannot be unpacked with the `...` operator
-$values = array_values($params);
-
-return new $class(...$values);
+return (new \ReflectionClass($class))->newInstanceArgs($params);
```

**UPDATE:** Apparently as of PHP 8.1 using the splat operator to pass named arguments *is supported*! See example 19 in https://www.php.net/manual/en/functions.arguments.php#example-505
So we don't need to use reflection at all.

## Caveats
- We name a lot of contructor args in injection yaml configuration - but those names don't always match up with real constructor argument names. We would need to review all of these and ensure that only real constructor argument names are used.
    - This is what makes it a breaking change.
- `InjectionCreator` is effectively a factory (it's invoked in `Injector::instantiate()` as [`$factory->create($class, $constructorParams)`](https://github.com/silverstripe/silverstripe-framework/blob/a1eee2a9ef78d04fa6558a89780b1ba0d72c5844/src/Core/Injector/Injector.php#L631)) - this means the above solution would only work when a different specific factory has _not_ been defined. There may therefore still be scenarios where named arguments are not respected.
   - tl;dr: In some of those scenarios it may result in silent failures where the argument is passed to the wrong parameter.
- ~~We swapped _away_ from using injection for dependency injection for a performance boost - see https://github.com/silverstripe/silverstripe-framework/pull/10265~~

**Comment by lozcalver** [comment]
It’s also worth noting that we switched from Reflection to using `new $class(...$values)` for a performance boost: https://github.com/silverstripe/silverstripe-framework/pull/10265

**Comment by GuySartorelli** [comment]
I'll add that to the caveats - that may be reason enough not to do this, but still worth discussing I reckon.

**Comment by GuySartorelli** [comment]
UPDATE: Apparently as of PHP 8.1 using the splat operator to pass named arguments *is supported*! See example 19 in https://www.php.net/manual/en/functions.arguments.php#example-505
